frontend/javascripts/oxalis/view/right-menu/trees_tab_view.js

```diff
@@ -661,98 +662,109 @@ class TreesTabView extends React.PureComponent<Props, State> {
-              </ButtonComponent>
-            </Tooltip>
-          </AdvancedSearchPopover>
-          <ButtonComponent onClick={this.props.onCreateTree} title="Create Tree">
-            <i className="fas fa-plus" /> Create
-          </ButtonComponent>
-          <ButtonComponent onClick={this.handleDelete} title="Delete Tree">
-            <i className="far fa-trash-alt" /> Delete
-          </ButtonComponent>
-          <ButtonComponent onClick={this.toggleAllTrees} title="Toggle Visibility of All Trees">
-            <i className="fas fa-toggle-on" /> Toggle All
-          </ButtonComponent>
-          <ButtonComponent
-            onClick={this.toggleInactiveTrees}
-            title="Toggle Visibility of Inactive Trees"
-          >
-            <i className="fas fa-toggle-off" /> Toggle Inactive
-          </ButtonComponent>
-          <Dropdown overlay={this.getActionsDropdown()} trigger={["click"]}>
-            <ButtonComponent>
-              More
-              <Icon type="down" />
-            </ButtonComponent>
-          </Dropdown>
-        </ButtonGroup>
-        <InputGroup compact>
-          <ButtonComponent onClick={this.props.onSelectNextTreeBackward}>
-            <i className="fas fa-arrow-left" />
-          </ButtonComponent>
-          <InputComponent
-            onChange={this.handleChangeTreeName}
-            value={activeTreeName || activeGroupName}
-            disabled={noTreesAndGroups}
-            style={{ width: "60%" }}
-          />
-          <ButtonComponent onClick={this.props.onSelectNextTreeForward}>
-            <i className="fas fa-arrow-right" />
-          </ButtonComponent>
-          <Dropdown overlay={this.getSettingsDropdown()} trigger={["click"]}>
-            <ButtonComponent title="Sort">
-              <i className="fas fa-sort-alpha-down" />
-            </ButtonComponent>
-          </Dropdown>
-        </InputGroup>
-        {noTreesAndGroups ? (
-          <Empty
-            image={Empty.PRESENTED_IMAGE_SIMPLE}
-            description={
-              <span>
-                There are no trees in this tracing.
-                <br /> A new tree will be created automatically once a node is set.
-              </span>
-            }
-          />
-        ) : (
-          <ul style={{ flex: "1 1 auto", overflow: "auto", margin: 0, padding: 0 }}>
-            <div className="tree-hierarchy-header">{this.getSelectedTreesAlert()}</div>
-            {this.getTreesComponents(orderAttribute)}
-          </ul>
-        )}
-        {groupToDelete !== null ? (
-          <DeleteGroupModalView
-            onCancel={this.hideDeleteGroupsModal}
-            onJustDeleteGroup={() => {
-              this.deleteGroupAndHideModal(groupToDelete, false);
-            }}
-            onDeleteGroupAndTrees={() => {
-              this.deleteGroupAndHideModal(groupToDelete, true);
-            }}
-          />
-        ) : null}
+        <DomVisibilityObserver targetId={treeTabId}>
```

Comment:
The trees tab view eats a considerable amount of performance for large tracings. However, I didn't find easy gains, so I went with this solution which should give some relief if one is willing to hide the trees tab for a moment.